hypervisor/qemu/qemu_test.go

```diff
@@ -9,24 +9,111 @@ package qemu
 
 import (
 	"testing"
+
+	. "github.com/mikelangelo-project/capstan/testing"
+	. "gopkg.in/check.v1"
 )
 
-var parsingtests = []struct {
-	in  string
-	out *Version
-}{
-	{"QEMU emulator version 1.0 (qemu-kvm-1.0), Copyright (c) 2003-2008 Fabrice Bellard", &Version{Major: 1, Minor: 0, Patch: 0}},
-	{"QEMU emulator version 1.6.2, Copyright (c) 2003-2008 Fabrice Bellard", &Version{Major: 1, Minor: 6, Patch: 2}},
-	{"QEMU PC emulator version 0.12.1 (qemu-kvm-0.12.1.2), Copyright (c) 2003-2008 Fabrice Bellard", &Version{Major: 0, Minor: 12, Patch: 1}},
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type suite struct{}
+
+var _ = Suite(&suite{})
+
+func (*suite) TestVersionParsing(c *C) {
+	m := []struct {
+		in  string
+		out *Version
+	}{
+		{"QEMU emulator version 1.0 (qemu-kvm-1.0), Copyright (c) 2003-2008 Fabrice Bellard", &Version{Major: 1, Minor: 0, Patch: 0}},
+		{"QEMU emulator version 1.6.2, Copyright (c) 2003-2008 Fabrice Bellard", &Version{Major: 1, Minor: 6, Patch: 2}},
+		{"QEMU PC emulator version 0.12.1 (qemu-kvm-0.12.1.2), Copyright (c) 2003-2008 Fabrice Bellard", &Version{Major: 0, Minor: 12, Patch: 1}},
+	}
+	for i, args := range m {
+		c.Logf("CASE #%d", i)
+
+		// This is what we're testing here.
+		version, err := ParseVersion(args.in)
+
+		// Expectations.
+		c.Check(err, IsNil)
+		c.Check(version, DeepEquals, args.out)
+	}
 }
 
-func TestVersionParsing(t *testing.T) {
-	for i, tt := range parsingtests {
-		version, err := ParseVersion(tt.in)
-		if err != nil {
-			t.Errorf("%d. ParseVersion(%q) => error %q, want %q", i, tt.in, err, tt.out)
-		} else if version.Major != tt.out.Major || version.Minor != tt.out.Minor || version.Patch != tt.out.Patch {
-			t.Errorf("%d. ParseVersion(%q) => %q, want %q", i, tt.in, version, tt.out)
+func (*suite) TestVmArguments(c *C) {
```

Comment:
Just checking: is this func meant to test other options in the future? Currently, it's very much focused on the volumes (apart from a simple default test).